xarray/backends/common.py

```diff
@@ -535,6 +535,17 @@ def open_datatree(
 
         raise NotImplementedError()
 
+    def open_groups(
+        self,
+        filename_or_obj: str | os.PathLike[Any] | BufferedIOBase | AbstractDataStore,
+        **kwargs: Any,
+    ) -> dict[str, Dataset]:
+        """
+        Backend open_groups method used by Xarray in :py:func:`~xarray.open_groups`.
+        """
```

Comment:
Some description of the motivation + relationship to `open_datatree` for developers would be good here, e.g.

```suggestion
        """
        Opens a dictionary mapping from group names to Datasets.
        
        Called by :py:func:`~xarray.open_groups`. This function exists so as to provide a universal way to open all groups in a file, before applying any additional consistency checks or requirements necessary to create a `DataTree` object (typically done using :py:meth:`~xarray.DataTree.from_dict`). 
        """
```